mark.c

```diff
@@ -841,18 +841,18 @@ GC_INNER mse * GC_mark_from(mse *mark_stack_top, mse *mark_stack,
         /* marking for that chunk.                                      */
         /* If the pointer is out of bounds at the upper-end, reset      */
         /* the pointer within bounds and continue from there            */
+        if (cheri_getaddress(limit) + sizeof(ptr_t) - 1
```

Comment:
I think we might be able to add clearer comments here. Perhaps something like "If the `limit` capability restricts us to reading fewer than `sizeof(ptr_t)` bytes a) that there can't possibly be a pointer at `limit`'s pointer b) that trying read `sizeof(ptr_t)` might cause a capability exception." before the `if` and "Decrement `limit` so that it's within `current_p`'s bounds" inside the `if`?